src/main/java/org/folio/service/pieces/flows/delete/PieceDeleteFlowManager.java

```diff
@@ -140,4 +143,26 @@ private Future<JsonObject> getOnOrderItemForPiece(Piece piece, RequestContext re
       return Future.succeededFuture();
     }
   }
+
+ public Future<Void> batchDeletePiece(List<String> ids, RequestContext requestContext) {
+   PieceDeletionHolder holder = new PieceDeletionHolder().withDeleteHolding(false);
+
+   List<Future> deleteFutures = ids.stream()
+     .map(id -> pieceStorageService.getPieceById(id, requestContext)
+       .map(pieceToDelete -> {
+         holder.withPieceToDelete(pieceToDelete);
+         return null;
+       })
+       .compose(aHolder -> basePieceFlowHolderBuilder.updateHolderWithOrderInformation(holder, requestContext))
```

Comment:
The same logic also is in deletePiece method. We should avoid code duplication, logic should be in 1 places

Reply:
thank you Serhii, I will rewrite this part